### PR TITLE
Fix in order parameter validation

### LINKED ISSUE #1443: In order parameter validation is broken
For example the following validation:

```java
validatorHelper.param.inOrder() //
				.type(boolean.class.getName()).optional() //
				.type(long.class.getName()).optional() //
				.type(int.class.getName()).optional() //
				.validate(executableElement, valid);
```

allows this method:
```java
void select(AdapterView<Adapter> a, long c, boolean b, Bundle bundle) { }
```

### PR DESCRIPTION
**Pull request by yDelouis**
Fixes #1443.

### CODE REVIEW

File: AndroidAnnotations/androidannotations-core/androidannotations/pom.xml [hunk 1]
```diff
@@ -56,6 +56,11 @@
 			<version>4.1.1.4</version>
 			<scope>provided</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
+			<version>1.10.8</version>
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/pom.xml]: Missing `test` scope!

Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/pom.xml]: Moreover, we use Mockito in other modules as well. So you can pull it to the root POM in `dependencyManagement`, just like we do with JUnit.

File: AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/helper/ValidatorParameterHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,321 @@
+package org.androidannotations.helper;
+
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/helper/ValidatorParameterHelperTest.java]: OK, this file should be reformatted.

**Comment by WonderCsabo** [comment]
The test suite is far from full coverage, but it is a good start. Thanks for this PR, we really needed this fix, and we can move on.

It seems Travis is down, so that is why it did not tell you have lots of Checkstyle errors in the test class and a missing licence header. Please fix this, as well as my other comments. After it, the PR can be merged.

**Comment by WonderCsabo** [comment]
We have a problem:

```
ItemClicksHandledActivity.java:[116,9] org.androidannotations.annotations.ItemSelect can only have the following parameters: [ [ boolean or java.lang.Boolean ], [ extending java.lang.Object (optional) ],  ] in the order above
ItemClicksHandledActivity.java:[131,9] org.androidannotations.annotations.ItemSelect can only have the following parameters: [ [ boolean or java.lang.Boolean ], [ extending java.lang.Object (optional) ],  ] in the order above
ItemClicksHandledActivity.java:[142,9] org.androidannotations.annotations.ItemSelect can only have the following parameters: [ [ boolean or java.lang.Boolean ], [ extending java.lang.Object (optional) ],  ] in the order above
```

The problem here that `ItemSelect` really accepts any type (any reference type or primitive type), except that if the second parameter is primitive int, than it is treated as a position, otherwise as the element.
But `anyType()` does not accept any types, only any classes. We should rename it to `anyClass()` or such, then introduce a new `anyType()` validator which indeed accepts any type (primitive or class).

File: AndroidAnnotations/androidannotations-core/androidannotations/pom.xml (hunk 1 above)
Comment (yDelouis) [inline AndroidAnnotations/androidannotations-core/androidannotations/pom.xml]: Fixed.

File: AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/helper/ValidatorParameterHelperTest.java (hunk 2 above)
Comment (yDelouis) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/helper/ValidatorParameterHelperTest.java]: I worked on my professional computer and forgot to setup eclipse formatter on it.

**Comment by yDelouis** [comment]
I addressed your comments.
I hope @dodgex is okay for setting him as author of the last commit since I copied-pasted his gist.

**Comment by dodgex** [comment]
@yDelouis sure i'm okay with that. Thanks :)

**Comment by WonderCsabo** [comment]
@yDelouis i just saw the `getWrapperType()` method, and it is half-done, not sure why. Can you fix that?

**Comment by WonderCsabo** [comment]
@yDelouis also please rebase this so the build can go green.

**Comment by WonderCsabo** [comment]
@yDelouis any update on this?

**Pull request closed without merging**